Revise modular boost policy and clarify options which do no impact `package_id` (#15494)
conan-io/conan-center-index: docs/faqs.md

## Can we add package which are parts of bigger projects like Boost?

> Sadly no. There have been many efforts in the past and we feel it's not sustainable given the number of combinations of libraries and version.

Reviewer: ```suggestion
Sadly no. There have been many efforts in the past and we feel it's not sustainable given the number of combinations of libraries and versions.
```

See #14660 for recent discussions. There is one main "boost" recipe with many versions maintained. Adding boost libraries with no dependencies

> just opens the does to graph resolution problems and once available allows for depedant libraries to be added.

Reviewer 2: ```suggestion
just opens the door to graph resolution problems and once available allows for dependent libraries to be added.
```

> In order to avoid this the sole permuation which permissible is when the project does not package any headers under the `boost/` folder, does not use the boost namespace

Reviewer: ```suggestion
In order to avoid this the sole permutation which is permissible is when the project does not package any headers under the `boost/` folder, does not use the boost namespace
```

and does not install libraries with the boost prefix.

### Can I add my project which I will submit to Boost?

Yes, but make sure it does not have Boost in the name. Use the [`author-name` convention](https://github.com/conan-io/conan-center-index/blob/master/docs/faqs.md#what-is-the-policy-on-recipe-name-collisions) so there are no conflicts. In addition to follow the rules outlined above.

## Can I add options that do not affect `package_id` or the package contents

> Generally no, these shorts of options can most likely be set from a profile or downstream recipes. However if the project supports this option from its build script

Reviewer: ```suggestion
Generally no, these sorts of options can most likely be set from a profile or downstream recipes. However if the project supports this option from its build script
```

> and would otherwise dynamically embeed this into the CMake config files or generated pkg-config files then it should be allowed.

Reviewer 2: ```suggestion
and would otherwise dynamically embed this into the CMake config files or generated pkg-config files then it should be allowed.
```

> Doing so requires [deleting the option from the `package_id`](adding_packages/conanfile_attributes.md#removing-from-package_id)

Reviewer: ```suggestion
Doing so requires [deleting the option from the `package_id`](adding_packages/conanfile_attributes.md#removing-from-package_id).
```
